ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java

```diff
@@ -117,57 +117,54 @@ public TaskResult run() {
     Histogram aggregationTimeMs =
         statsReportType == StatsReportType.ACCOUNT_REPORT ? metrics.accountStatsAggregationTimeMs
             : metrics.partitionClassStatsAggregationTimeMs;
-    synchronized (accountStatsStore) {
-      long startTimeMs = System.currentTimeMillis();
-      try {
-        List<String> instanceNames = manager.getClusterManagmentTool().getInstancesInCluster(manager.getClusterName());
-        Map<String, StatsWrapper> statsWrappers =
-            statsReportType == StatsReportType.ACCOUNT_REPORT ? fetchAccountStatsWrapperForInstances(instanceNames)
-                : fetchPartitionClassStatsWrapperForInstances(instanceNames);
-        fetchTimeMs.update(System.currentTimeMillis() - startTimeMs);
-        logger.info("Aggregating stats from " + statsWrappers.size() + " hosts");
-        results = clusterAggregator.doWorkOnStatsWrapperMap(statsWrappers, statsReportType);
-        if (statsReportType == StatsReportType.ACCOUNT_REPORT) {
-          if (clusterMapConfig.clustermapEnableDeleteInvalidDataInMysqlAggregationTask) {
-            removeInvalidAggregatedAccountAndContainerStats(results.getSecond());
-          }
-          accountStatsStore.storeAggregatedAccountStats(results.getSecond());
-        } else if (statsReportType == StatsReportType.PARTITION_CLASS_REPORT) {
-          if (clusterMapConfig.clustermapEnableDeleteInvalidDataInMysqlAggregationTask) {
-            removeInvalidAggregatedPartitionClassStats(results.getSecond());
-          }
-          accountStatsStore.storeAggregatedPartitionClassStats(results.getSecond());
+    long startTimeMs = System.currentTimeMillis();
```

Comment:
Is the only change here the removal of synchronization and the close connection at the end?

Was the synchronization before to protect the sql connection related data structures, but that is no longer required since the connection pool now handles it?

Reply:
yes, that's the only change for these files. 

We used to have only one connection, now with the connection pool can we more than one connections, we don't need to synchronize those block anymore.